.github/README.md

```diff
@@ -0,0 +1,137 @@
+Chipyard Continuous Integration (CI)
+===========
+
+Website: https://gihub.com/gh/ucb-bar/chipyard/actions
+
+GitHub Actions Brief Explanation
+---------------------------
+
+CI is executed by Github Actions (GA). GA is controlled by `.yml` files in the `.github/workflows/` directory. 
+In our case we have just one workflow named `chipyard-rocket-run-tests.yml`.
+It defines a number of `jobs` within it that do particular tasks.
+All jobs in the workflow must pass for the CI run to be successful.
+In general, a job is run in parallel with others unless it depends on some other job.  
+The dependency of one job on the completion of another is specified via the `needs` field.
+
+For example:
+```yaml
+  prepare-chipyard-cores:
+    name: prepare-chipyard-cores
+    needs: [make-keys, setup-complete]
+```
+This specifies that the `prepare-chipyard-cores` job needs the both the `make-keys` and the `setup-complete` steps to
+be completed before it can run.
+
+Chipyard runs its CI using a docker image created from `dockerfiles/Dockerfile`.
+See its [README](../dockerfiles/README.md) for more details.
+
+Finally, within each job's `steps:` section, the steps are run sequentially and state persists throughout a job.
+So when you run something like `checkout` the next step has the checked out code.
+
+[Composite Actions](https://docs.github.com/en/actions/creating-actions) (CA) allow for limited subroutine like code re-use within GA.
+We use both community created and our own Composite Actions in our CI process. CA capabilities are changing rapidly.
+Nesting of composite actions was only recently unveiled. There is a lot of room for more code reuse, in particular
+we specify things over and over like docker image tag and checkout commands.
+
+One use of CA: our process relies on caching to avoid running time-consuming and intensive tasks more often than necessary.
+
+The following is an example of using the cache@v2 composite action. A step `uses: actions/cache@v2` which take as parameters the
+path that contains the data to be cached and a key. Paths can have multiple targets. 
+The following step can look at the result of the cache operation, if there was cache miss, then we run the command that
+will generate the data to be cached. The caching of the generated data is implicit.
+>Note: GA cache documentation suggests using the yml level `if: steps.cache-primes.outputs.cache-hit != 'true'` to
+> determine whether to run the data generation command.
+> At the time of this writing the if construct has a bug and will not run correctly within a composite action. The use
+> of a bash based if is a hack found on stackoverflow
```

Comment:
Can you post the link to the StackOverflow just in case we need to fix this up later?

Reply:
Done